src/consensus/pbft/libbyz/replica.cpp

```diff
@@ -2006,20 +2063,23 @@ void Replica::process_new_view(Seqno min, Digest d, Seqno max, Seqno ms)
       {
         if (ledger_writer && req_in_pp > 0)
         {
+          pp->set_view(prev_view);
           last_te_version = ledger_writer->write_pre_prepare(pp);
+          pp->set_view(v);
         }
       }
     }
     else
     {
       ByzInfo info;
       pc.add_old(pp);
-
       if (execute_tentative(pp, info))
       {
         if (ledger_writer && req_in_pp > 0)
         {
+          pp->set_view(prev_view);
           last_te_version = ledger_writer->write_pre_prepare(pp);
+          pp->set_view(v);
```

Comment:
it seems like the view should be pass to write_pre_prepare and it can be responsible for setting the view on the pre-prepare?

Reply:
I think it should be clear that we are setting the view during view change here instead of having `write_pre_prepare` hide that detail? Also I think that `write_pre_prepare` should just write things to the ledger and not implement any more logic than it needs to.

Reply:
The issue here is that this logic is repeated twice and is not clear what it does by just glancing at it.

And alternative is create a wrapper function.

Reply:
fair enough :)

Reply:
Done